### PR TITLE
Update to python 3.9 with pyupgrade

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -584,14 +584,7 @@ class AccessorCallableDocumenter(AccessorLevelDocumenter, MethodDocumenter):
...
     def format_name(self):
-        if sys.version_info < (3, 9):
-            # NOTE pyupgrade will remove this when we run it with --py39-plus
-            # so don't remove the unnecessary `else` statement below
-            from pandas.util._str_methods import removesuffix
-
-            return removesuffix(MethodDocumenter.format_name(self), ".__call__")
-        else:
-            return MethodDocumenter.format_name(self).removesuffix(".__call__")
+        return MethodDocumenter.format_name(self).removesuffix(".__call__")
```
Comment (MarcoGorelli) [inline doc/source/conf.py:587]: been waiting for this to happen 🎉 🧙

File: pandas/_testing/__init__.py [hunk 2]
```diff
@@ -923,7 +925,6 @@ def external_error_raised(expected_exception: type[Exception]) -> ContextManager
     Callable
         Regular `pytest.raises` function with `match` equal to `None`.
     """
-    import pytest
```
Comment (MarcoGorelli) [inline pandas/_testing/__init__.py:926]: why does this one need moving?

Comment (phofl) [reply pandas/_testing/__init__.py:926]: Good question, pyupgrade removed it automatically. Lets move it back and see what happens

File: pandas/_config/config.py [hunk 3]
```diff
@@ -70,6 +69,9 @@
 from pandas._typing import T  # noqa: TCH001
 from pandas.util._exceptions import find_stack_level
 
+if TYPE_CHECKING:
+    from collections import abc
```
[the remark this answers is not in the document]
Comment (phofl) [reply pandas/_config/config.py:73]: We have a check that disallow this pattern.

[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply pandas/_config/config.py:73]: it was added in https://github.com/pandas-dev/pandas/pull/25957/files

tbh i'd be in favour of removing, it seems pretty pointless

[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply pandas/_config/config.py:73]: anyway, looks like there's a failing check, but other than that it looks ready

up to you if you want to remove the other check and reduce the diff

**Comment by mroeschke** [comment]
Feel free to reopen when you have time to circle back here

**Pull request closed without merging**